pkgs/unified_analytics/lib/src/analytics.dart

```diff
@@ -235,18 +235,28 @@ class AnalyticsImpl implements Analytics {
       initializer: initializer,
     );
 
-    // Initialize the config handler class and check if the
-    // tool message and version have been updated from what
-    // is in the current file; if there is a new message version
-    // make the necessary updates
-    if (!_configHandler.parsedTools.containsKey(tool)) {
-      _configHandler.addTool(tool: tool);
+    // If the tool has already been added to the config file
+    // we can assume that the client has successfully shown
+    // the consent message
+    //
+    // Additionally, if the tool does not exist in the config
+    // file but the config file already exists, prompt with
+    // message again
+    if (_configHandler.parsedTools.containsKey(tool.label)) {
+      _clientShowedMessage = true;
+    } else {
       _showMessage = true;
     }
-    if (_configHandler.parsedTools[tool]!.versionNumber < toolsMessageVersion) {
-      _configHandler.incrementToolVersion(tool: tool);
+
+    // Check if the tool has already been onboarded, and if it
+    // has, check if the latest message version is greater to
+    // prompt the client to show a message
+    if (_configHandler.parsedTools.containsKey(tool.label) &&
+        _configHandler.parsedTools[tool.label]!.versionNumber <
```

Comment:
Consider
```
var version = _configHandler.parsedTools[tool.label]?.versionNumber ?? -1;
if (version < toolsMessageVersion) ...
```
That removes an unnecessary map lookup, which will improve the performance.

Reply:
Love that, simplified the logic above it as well where we were checking if the tool existed in the config file